airflow/providers/google/cloud/links/automl.py

```diff
@@ -0,0 +1,161 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains Google AutoML links."""
+from typing import TYPE_CHECKING, Optional
+
+from airflow.providers.google.cloud.links.base import BaseGoogleLink
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+AUTOML_BASE_LINK = "https://console.cloud.google.com/automl-tables"
+AUTOML_DATASET_LINK = (
+    AUTOML_BASE_LINK + "/locations/{location}/datasets/{dataset_id}/schemav2?project={project_id}"
+)
+AUTOML_DATASET_LIST_LINK = AUTOML_BASE_LINK + "/datasets?project={project_id}"
+AUTOML_MODEL_LINK = (
+    AUTOML_BASE_LINK
+    + "/locations/{location}/datasets/{dataset_id};modelId={model_id}/evaluate?project={project_id}"
+)
+AUTOML_MODEL_TRAIN_LINK = (
+    AUTOML_BASE_LINK + "/locations/{location}/datasets/{dataset_id}/train?project={project_id}"
+)
+AUTOML_MODEL_PREDICT_LINK = (
+    AUTOML_BASE_LINK
+    + "/locations/{location}/datasets/{dataset_id};modelId={model_id}/predict?project={project_id}"
+)
+
+
+class AutoMLDatasetLink(BaseGoogleLink):
+    """Helper class for constructing AutoML Dataset link"""
+
+    name = "AutoML Dataset"
+    key = "automl_dataset"
+    format_str = AUTOML_DATASET_LINK
+
+    @staticmethod
+    def persist(
+        context: "Context",
+        task_instance,
+        dataset_id: str,
+        project_id: Optional[str],
```

Comment:
Looking at all of the URL formats, it seems like `project_id` is required. Is that true? If so, the typing should be updated to reflect this throughout.

Reply:
@josh-fell you are right this parameter is required for links, but for old google operators the project_id is optional. We solve this problem by adding this code `self.project_id or hook.project_id`, but these parameters are optional too and when I make project_id in links required our static checks fail. It is the reason why we make this parameter Optional

Reply:
Thanks for the context! So thinking about this another way, it seems like mypy found an edge case where it's _possible_ to not pass a value for `project_id` when the `persist()` methods get called via the typing of the input args. Instead of trying to make mypy happy, it might be worthwhile adding a check ensuring a value is present to pass in as a `project_id` arg and having the typing of `project_id` in the operator link reflect that it _is_ required. WDYT?

Reply:
@josh-fell I think it is good idea. I have changed the code in last commit. Now the `project_id` is required for links.